version: "3.9"
services:
  app-api:
    build:
      context: .
      dockerfile: Dockerfile
      target: local
    depends_on:
      app-mysql:
        condition: service_healthy
    container_name: app-api
    restart: unless-stopped
    tty: true
    ports:
      - "3323:3323"
      - "5555:5555"
    volumes:
      - .:/usr/src/app/
    networks:
      - api-network
    command: npm run api:dev
  app-mysql:
    image: mysql:8
    container_name: app-mysql
    restart: unless-stopped
    tty: true
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
    healthcheck:
      test: mysqladmin ping -h 127.0.0.1 -u ${MYSQL_USER} --password=${MYSQL_PASSWORD}
      interval: 3s
      timeout: 5s
      retries: 10
    volumes:
      - app-mysql-data:/var/lib/mysql/
      - ./prisma/mysql:/docker-entrypoint-initdb.d/
    networks:
      - api-network

networks:
  api-network:
    driver: bridge

volumes:
  app-mysql-data:
    driver: local
